api.go

```diff
@@ -86,14 +86,32 @@ func (api *CacophonyAPI) JustRegistered() bool {
 	return api.justRegistered
 }
 
-// NewAPIFromConfig prases the supplied configFile and creates a new CacophonyAPI with the configFile information
+// NewAPIFromConfig parses file name into deviceConfigPath if it doesn't exist or overwrite set
+// and creates a new CacophonyAPI with the configFile information
 // and saves the generated password to privConfigFileName(configFile)
-func NewAPIFromConfig(configFile string) (*CacophonyAPI, error) {
-	conf, err := ParseConfigFile(configFile)
+func NewAPIFromConfig(filename string, overwrite bool) (*CacophonyAPI, error) {
```

Comment:
How would the caller decide what to pass for the `overwrite` value? Would it better to always update device.yaml?